v0/destinations/delighted/transform.js

```diff
@@ -0,0 +1,225 @@
+    getErrorRespEvents,
+    constructPayload,
+    getSuccessRespEvents,
+    getDestinationExternalID,
+    isEmptyObject
+} = require("../../util");
+
+const {
+    channelValidity,
+    eventValidity,
+    isValidEmail,
+    isValidPhone,
+    userValidity
+} = require("./util");
+const {
+    ENDPOINT,
+    DELIGHTED_EXCLUSION_FIELDS,
+    identifyMapping
+} = require("./config");
+
+const identifyResponseBuilder = async (message, {Config}) => {
+    const userId = getFieldValueFromMessage(message, "userIdOnly");
+    if (!userId) {
+      throw new CustomError(
+        "userId is required for identify",
+        400
+      );
+    }
+    let channel = getDestinationExternalID(message, "delightedChannelType") || Config.channel;
+    channel = channel.toLowerCase();
+
+    channel= channelValidity(channel,userId);
+    let payload = constructPayload(message, identifyMapping);
+
+    payload.send = false;
+    payload.channel = channel;
+    payload.delay = Config.delay || message.context.traits.delay || 0 ;
+    payload.last_sent_at = message.lastSentAt;
+
+    if(!payload.name){
+        const fName = getFieldValueFromMessage(message, "firstName" );
+        const lName = getFieldValueFromMessage(message, "lastName");
+        let name= fName.concat(lName);
+        payload.name = name ? name : userId;
+    }
+    
+    let properties = {};
+    properties = extractCustomFields(
+        message,
+        properties,
+        ["context.traits"],
+        DELIGHTED_EXCLUSION_FIELDS
+    );
+    payload = {
+        ...payload,
+        properties
+    };
+    //update/create the user
+    const response = defaultRequestConfig();
+    response.headers = {
+        "Authorization": Config.apiKey,
+        "Content-Type":  "application/json"
+    }
+    response.method = defaultPostRequestConfig.requestMethd;
+    response.endpoint=`${ENDPOINT}`; // since ENDPOINT remains same
+    response.body.JSON = removeUndefinedAndNullValues(payload);
+    return response;
+};
+
+const trackResponseBuilder = async (message, {Config}) => {
+
+    //checks if the event is valid if not throws error else nothing
+    eventValidity(Config,message);
+    //getting the userId
+    const userId = getFieldValueFromMessage(message, "userIdOnly");
+    if (!userId) {
+      throw new CustomError(
+        "userId is required for identify",
+        400
+      );
+    }
+    let channel = getDestinationExternalID(message, "delightedChannelType") || Config.channel;
+    channel = channel.toLowerCase();
+
+    channel= channelValidity(channel,userId);
+    
+    //checking if user already exists or not, throw error if it doesn't
+    let check = await userValidity(channel, Config);
+    if(!check){
+        throw new CustomError(
+            `user ${userId} doesnot exist`,
+            400
+        );
+    }
+    let payload = {};
+
+    payload.send = true;
+    payload.channel = channel;
+    payload.delay = Config.delay || message.context.traits.delay || 0 ;
```

Comment:
let's no put `delay` in `context.traits`. For track call, we can maybe allow it to be passed in `properties`